compiler/src/iree/compiler/GlobalOptimization/RaiseSpecialOps.cpp

```diff
@@ -676,6 +680,95 @@ static Value rewriteCatNegateAndSlice(RewriterBase &rewriter,
+    // of a change.
+    Type outElementType = getElementTypeOrSelf(namedOp->getResultTypes()[0]);
+    bool didChangeOperand = false;
+    {
+      OpBuilder::InsertionGuard guard(rewriter);
+      Block *block = &namedOp->getRegion(0).front();
+      rewriter.setInsertionPointToStart(block);
+      auto replaceOperandWithTypeCast = [&](OpOperand &operand) {
+        // If the op already has implicit casting semantics for this operand,
+        // do not fuse.
+        if (getElementTypeOrSelf(operand.get().getType()) != outElementType) {
+          return false;
+        }
+        auto producer = operand.get().getDefiningOp<linalg::GenericOp>();
+        if (!producer) {
+          return false;
+        }
+        if (!linalg::isElementwise(producer) ||
+            producer.getNumDpsInputs() != 1 || producer.getNumDpsInits() != 1) {
+          return false;
+        }
+
+        if (!llvm::hasSingleElement(
+                producer.getBlock()->without_terminator())) {
+          return false;
+        }
+        // We only handle arith.extf here for two reasons:
+        //  1) This pattern is being applied to convolution/contraction
+        //     interfaces. Extension semantics for integers depend on the named
+        //     op and requires a slightly different pattern.
+        //  2) Truncating operations like `arith.truncf` should not be fused
+        //     with consumers; it would be preferred to fuse those with
+        //     producers (and the consumer fusion is arguably the less canonical
+        //     form).
+        if (!llvm::isa<arith::ExtFOp>(
+                *producer.getBlock()->without_terminator().begin())) {
+          return false;
+        }
+        Type producerElementType = getElementTypeOrSelf(
+            producer.getDpsInputOperand(0)->get().getType());
+        int64_t operandNumber = operand.getOperandNumber();
+        // Set the operand to the linalg op to the smaller one.
+        namedOp->setOperand(operandNumber, producer->getOperand(0));
+
+        // Insert a new block argument into the body of the named op with the
+        // correct type.
+        Value blockArg = block->insertArgument(
+            operandNumber, producerElementType, namedOp.getLoc());
+        // Create the extf.
+        auto ext = rewriter.create<arith::ExtFOp>(namedOp.getLoc(),
+                                                  outElementType, blockArg);
+        // Replace uses of the old argument with the extended value.
+        rewriter.replaceAllUsesWith(block->getArgument(operandNumber + 1),
+                                    ext.getResult());
+        // Erase the old argument.
+        block->eraseArgument(operandNumber + 1);
+        return true;
+      };
+
+      didChangeOperand = replaceOperandWithTypeCast(namedOp->getOpOperand(0));
+      didChangeOperand |= replaceOperandWithTypeCast(namedOp->getOpOperand(1));
+    }
+    if (didChangeOperand) {
+      return success();
+    }
+    return failure();
```

Comment:
This can be `return success(didChangeOperand)`.

Reply:
Cool! I didn't know this existed.